pages/common.js

```diff
@@ -0,0 +1,182 @@
+/**
+ * NOTE: This file contains supplementary code shared by both "shows page" and "episodes page"
+ */
+
+// create some initial stage static page html content
+function createInitialContent(type) {
+  const rootElem = document.getElementById("root");
+  if (type === "shows") {
+    rootElem.innerHTML = initialShowsPageContent;
+  } else if (type === "episodes") {
+    rootElem.innerHTML = initialEpisodesContent;
+  } else {
+    rootElem.innerHTML = errorContent;
+  }
+}
+
+// set initial styling to some page components
+function setInitialStyling(pageName) {
+  const contentDiv = document.querySelector(`.all-${pageName}`);
+  const thumbnails = contentDiv.childNodes;
+  document.getElementById("txt-search").style.order = 1;
+  document.getElementById("txt-search").hidden = true;
+  document.getElementById("btn-search").style.order = 2;
+  document.getElementById("btn-search").classList.add("blue");
+
+  thumbnails.forEach((thumbnail) => {
+    thumbnail.querySelector("div:first-of-type").classList = "summary";
+    if (pageName === "shows") {
+      thumbnail.classList = "show thumbnail";
+      thumbnail.querySelector("h4").classList = "title show-title link";
+    } else {
+      thumbnail.classList = "episode thumbnail";
+      thumbnail.querySelector("h4").classList = "title episode-title";
+    }
+  });
+}
+
+// attach event listeners to elements
+function createEventListeners(pageName) {
```

Comment:
I like that you have created this function. makes your code clean and compact

Reply:
Thank you for the encouraging comment @SenaitEkubai. I appreciate it.